collector/src/main.rs

```diff
@@ -995,113 +996,66 @@ fn main_result() -> anyhow::Result<i32> {
             bench_rustc,
             self_profile,
         } => {
-            println!("processing commits");
+            println!("processing artifacts");
             let client = reqwest::blocking::Client::new();
-            let response: collector::api::next_commit::Response = client
-                .get(&format!("{}/perf/next_commit", site_url))
+            let response: collector::api::next_artifact::Response = client
+                .get(&format!("{}/perf/next_artifact", site_url))
                 .send()?
                 .json()?;
-            let next = if let Some(c) = response.commit {
+            let next = if let Some(c) = response.artifact {
                 c
             } else {
-                println!("no commit to benchmark");
-                // no missing commits
+                println!("no artifact to benchmark");
+                // no missing artifacts
                 return Ok(0);
             };
 
             let pool = database::Pool::open(&db.db);
 
-            let sysroot = Sysroot::install(next.commit.sha.to_string(), &target_triple)
-                .with_context(|| format!("failed to install sysroot for {:?}", next.commit))?;
-
-            let mut benchmarks = get_benchmarks(
-                &benchmark_dir,
-                next.include.as_deref(),
-                next.exclude.as_deref(),
-            )?;
-            benchmarks.retain(|b| b.category().is_primary_or_secondary());
+            if let NextArtifact {
+                artifact: ArtifactId::Tag(tag),
+                ..
```

Comment:
We should match on the enum/struct to avoid silently ignoring future artifact IDs if we add them.